.github/workflows/release.yml

```diff
@@ -68,68 +118,170 @@ jobs:
-      - name: Get Shortened Sha
-        id: sha
-        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+          path: build/libs/Darwin/x64/Release
+
+      - name: Get Windows x64 Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: win-x64
+          path: build/libs/Windows/x64/Release
+
+      - name: Update version number
+        run: dotnet cake ./scripts/build.cake -t AssignVersion --newVersion ${{needs.get_version_code.outputs.version}}
+
       - name: Build and Package
         working-directory: bindings/netstandard/ElectionGuard/ElectionGuard.Encryption
         run: |
           dotnet build -c Release
-          dotnet pack -c Release --version-suffix "v${{ steps.sha.outputs.sha_short }}"
-      - name: Get Release Version
-        id: get_version
-        uses: greenygh0st/net-proj-release-version@v2
-        with:
-          PROJ_FILE: ./bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/ElectionGuard.Encryption.csproj
-      - name: Release
-        uses: softprops/action-gh-release@v1
-        with:
-          name: Release ${{ steps.get_version.outputs.RELEASE_VERSION }}
-          tag_name: v${{ steps.get_version.outputs.RELEASE_VERSION }}
-          draft: false
-          prerelease: false
-          files: |
-            ./bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/bin/Release/*.nupkg
-            ./build/libs/Darwin/x64/Release/src/*.dylib
-            ./build/libs/Linux/x64/Release/src/*.so
-            ./build/libs/Windows/x64/Release/src/Release/*.dll*
-            ./build/libs/Windows/x64/Release/libs/hacl/Release/*.dll*
-      - name: Push to Nuget.org
+          dotnet pack -c Release --version-suffix ${{needs.get_version_code.outputs.build_number}}
+
+      - name: Publish to Nuget.org
         working-directory: bindings/netstandard/ElectionGuard/ElectionGuard.Encryption
         run: dotnet nuget push "**/*.nupkg" --api-key ${{ secrets.NUGET_API_KEY}} --source https://api.nuget.org/v3/index.json --skip-duplicate
+
+  publish_npm:
+    name: Publish NPM
+    needs: [get_version_code, wasm_build]
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+      - name: Install Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+      - name: Get WASM Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: wasm
+          path: bindings/typescript/src/wasm
+      - name: Install Dependencies
+        working-directory: bindings/typescript
+        run: npm install
+      - name: Update version number
+        working-directory: bindings/typescript
+        run: npm version ${{needs.get_version_code.outputs.version}} --no-git-tag-version
       - name: Publish to NPM
         working-directory: bindings/typescript
         run: |
           echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_REGISTRY_TOKEN }}" > .npmrc
```

Comment:
we can swap out the keys whenever, but we need ot be sure too update the package name as well